app/views.py

```diff
@@ -1,110 +1,277 @@
 """
 this files contains the logic and the routes of the app
 """
-from flask import request, url_for, abort, g, jsonify, session
+import re
+from flask import request, url_for, abort, session
 from flask_api import status, exceptions
+from flasgger import Swagger
+from flasgger.utils import swag_from
 
 from app import app
 from . models import Users, Events
+swagger = Swagger(app)
 
-# events = {}
-# Users = {}
 
-@app.route('/api/auth/register', methods=['POST'])
+@app.route('/api/auth/register/', methods=['POST'])
 def registration():
+    """user registration endpoint
+          registers a user and takes in a name, email and password
+     ---
+        tags:
+          - Bright Events API
+        parameters:
+          - in: formData
+            name: name
+            type: string
+            required: true
+            description: Enter a name to register!
+          - in: formData
+            name: email
+            type: string
+            required: true
+            description: The email address you will use to log in!
+          - in: formData
+            name: password
+            type: string
+            required: true
+            description: The password you will use to registert!
+        responses:
+          201:
+            description: User has been created successfully
+            schema:
+              $ref: '#/definitions/Task'
+            examples:
+                name: John Doe
+                email: test@test.com
+                password: password
     """
-    This function handles the user registration
-    """
-    username = request.data.get('username')
+    if "user" in session:
+        return {"message": "you are already logged in"}
+
+    # Retreive data from the user side
+    fname = request.data.get('first_name')
+    lname = request.data.get('last_name')
+    email = request.data.get('email')
     password = request.data.get('password')
 
-    if username is None or password is None:
-        abort(400)
-    user = Users(username, password)
-    if username in Users.user_db.keys():
-            return "User already exists. Please login.", status.HTTP_202_ACCEPTED
-    cur_user = user.save()
+    """
+    validate the data from user isalpha ensures there are no
+    non-alphabet charcters
+    """
+    if fname is None or fname.strip == "" or not fname.isalpha():
+        message = {"message": "check if the first name is correctly filled"}
+        return message, status.HTTP_400_BAD_REQUEST
+
+    if lname is None or lname.strip == "" or not lname.isalpha():
+        message = {"message": "check if the last name is correctly filled"}
+        return message, status.HTTP_400_BAD_REQUEST
 
-    return {'username': cur_user.username}, status.HTTP_201_CREATED
+    if email is None or email.strip == "" or not re.search(
+            r'[\w.-]+@[\w.-]+.\w+', email):
+        message = {"message": "The email address is not filled out correctly"}
+        return message, status.HTTP_400_BAD_REQUEST
```

Comment:
Validations mixed up. It would be a good idea to separate them and provide message for each differently.
```message = {"message": "The email address is not filled out correctly"}```
This would be general for an invalid email and an empty email field.

Reply:
noted